src/tests/Cities.test.js

```diff
@@ -0,0 +1,39 @@
+import { test, expect, jest } from '@jest/globals';
+import '@testing-library/jest-dom';
+import { render } from "@testing-library/react";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import Cities from "../components/Cities";
+import { configureStore } from "@reduxjs/toolkit";
+import rootReducer from "../reducers/cities/citiesSlice";
+
+test("renders cities correctly", () => {
+  const mockCities = [
+    { name: "City 1", population: 100000, image: "city1.jpg" },
+    { name: "City 2", population: 200000, image: "city2.jpg" },
+  ];
+
+  const store = configureStore({
+    reducer: rootReducer,
+    preloadedState: {
+      cities: { cities: mockCities, isLoading: false },
+    },
+  });
+
+  const mockNavigate = jest.fn();
+  jest.mock("react-router-dom", () => ({
+    ...jest.requireActual("react-router-dom"),
+    useNavigate: () => mockNavigate,
+  }));
+
+  const { getByText } = render(
+    <Provider store={store}>
+      <MemoryRouter>
+        <Cities />
+      </MemoryRouter>
+    </Provider>
+  );
+
+  expect(getByText("City 1")).toBeInTheDocument();
+  expect(getByText("City 2")).toBeInTheDocument();
+});
```

Comment:
- [ ] Dear @davidolanrewaju, Good job writing the tests here. But I figure you only have this and the other two snapshot tests. Please remember tests are a great tool to keep our functionalities in place and assert everything is working well. Please add some tests for the pure functions in the Redux code;
- [Optional] I can see you have made use of react testing library, But I would love to suggest adding more tests. This will give you the freedom of changing things in the future considering your test will fail if you break any functionalities.